modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesRecoveryTask.java

```diff
@@ -23,129 +23,197 @@
-        List<AndesQueue> fromDBDuplicated = new ArrayList<AndesQueue>(fromDB);
-
-        fromDB.removeAll(fromMap);
-        for (AndesQueue queue : fromDB) {
-            for (QueueListener listener : queueListeners) {
-                log.warn("Recovering node. Adding queue " + queue.toString());
-                listener.handleClusterQueuesChanged(queue, QueueListener.QueueEvent.ADDED);
-            }
-        }
-
-        fromMap.removeAll(fromDBDuplicated);
-        for (AndesQueue queue : fromMap) {
-            for (QueueListener listener : queueListeners) {
-                log.warn("Recovering node. Removing queue " + queue.toString());
-                listener.handleClusterQueuesChanged(queue, QueueListener.QueueEvent.DELETED);
-            }
-        }
-    }
-
-    private void reloadBindingsFromDB() throws AndesException {
-        List<AndesExchange> exchanges = andesContextStore.getAllExchangesStored();
-        for (AndesExchange exchange : exchanges) {
-            List<AndesBinding> fromDB = andesContextStore.getBindingsStoredForExchange(exchange.exchangeName);
-            List<AndesBinding> fromMap = amqpConstructStore.getBindingsForExchange(exchange.exchangeName);
-            List<AndesBinding> fromDBDuplicated = new ArrayList<AndesBinding>(fromDB);
-            fromDB.removeAll(fromMap);
-            for (AndesBinding binding : fromDB) {
-                for (BindingListener listener : bindingListeners) {
-                    log.warn("Recovering node. Adding binding " + binding.toString());
-                    listener.handleClusterBindingsChanged(binding, BindingListener.BindingEvent.ADDED);
-                }
-            }
-
-            fromMap.removeAll(fromDBDuplicated);
-            for (AndesBinding binding : fromMap) {
-                for (BindingListener listener : bindingListeners) {
-                    log.warn("Recovering node. removing binding " + binding.toString());
-                    listener.handleClusterBindingsChanged(binding, BindingListener.BindingEvent.DELETED);
-                }
-            }
-        }
-    }
-
-    private void reloadSubscriptions() throws AndesException {
-        ClusterResourceHolder.getInstance().getSubscriptionManager().reloadSubscriptionsFromStorage();
-    }
+public class AndesRecoveryTask implements Runnable, StoreHealthListener {
+
+	private List<QueueListener> queueListeners = new ArrayList<>();
+	private List<ExchangeListener> exchangeListeners = new ArrayList<>();
+	private List<BindingListener> bindingListeners = new ArrayList<>();
+
+	AndesContextStore andesContextStore;
+	AMQPConstructStore amqpConstructStore;
+
+	// set storeOperational to true since it can be assumed that the store is operational at startup
+	// if it is non-operational, the value will be updated immediately
+	AtomicBoolean isContextStoreOperational = new AtomicBoolean(true);
+
+	private static Log log = LogFactory.getLog(AndesRecoveryTask.class);
```

Comment:
Can we make the log variable final? Or else, it could be modified from some where else.